thor/src/lib.rs

```diff
@@ -144,22 +143,113 @@ impl Channel {
     /// Update the distribution of coins in the channel.
     ///
     /// It assumes that the counterparty has already agreed to update the
-    /// channel with the `new_balance` and the same `timelock` and will call
-    /// the same API (or an equivalent one).
+    /// channel with the same balance and `timelock` and will call the same API
+    /// (or an equivalent one).
     ///
     /// Consumers should implement the traits `SendMessage` and `ReceiveMessage`
     /// on the `transport` they provide, allowing the parties to communicate
     /// with each other.
     pub async fn update<T>(
         &mut self,
         transport: &mut T,
-        new_balance: Balance,
+        Balance { ours, theirs }: Balance,
+        time_lock: u32,
+    ) -> anyhow::Result<()>
+    where
+        T: SendMessage + ReceiveMessage,
+    {
+        let split_output_self = SplitOutput::Balance {
+            amount: ours,
+            address: self.final_address_self.clone(),
+        };
+
+        let split_output_other = SplitOutput::Balance {
+            amount: theirs,
+            address: self.final_address_other.clone(),
+        };
+
+        self._update(
+            transport,
+            vec![split_output_self, split_output_other],
+            time_lock,
+        )
+        .await
+    }
+
+    /// Perform an atomic swap with a thor channel as beta ledger in the
+    /// role of Alice.
+    pub async fn add_ptlc_alice<T>(
+        &mut self,
+        _transport: &mut T,
+        _new_split_outputs: Vec<SplitOutput>,
+        _encryption_secret_key: Scalar,
+        _alpha_absolute_expiry: u32,
+        _TX_s_time_lock: u32,
+        _ptlc_redeem_time_lock: u32,
+    ) -> anyhow::Result<()>
+    where
+        T: SendMessage + ReceiveMessage,
+    {
+        // TODO: Think about how to handle the three expiries.
+
+        // 1. Construct and sign ptlc_refund and ptlc_redeem transactions (this requires
+        // building TX_c and TX_s too).
+        // 2. Send signatures to Bob.
+        // 3. Caller must convince Bob that the alpha asset is funded (outside of this
+        // function).
+        // 4. Receive Bob's signature for ptlc_refund and encsignature for ptlc_redeem.
+        // 5. Run channel update protocol to add PTLC output.
+        // 6. Send encryption_secret_key to Bob (who will use it to redeem alpha asset).
+        // 7. Attempt to perform a channel update to add PTLC output to Alice's balance
+        // output. If Bob doesn't respond, force close and publish ptlc_redeem. If he
+        // does respond, carry out the update and feel free to forget the
+        // encryption_secret_key.
```

Comment:
What do you think of adding a second protocol/function for this? I.e. `add_ptlc_alice` only adds the PTLC to `Tx_s` and `redeem_ptlc` _redeems_ the PTLC off-chain.

Reply:
As discussed, since it's already correctly implemented as one API which makes sense as a whole we leave it as it is. But eventually we could split it up if it proves useful/necessary.